src/lib/coda_networking/coda_networking.ml

```diff
@@ -446,9 +449,53 @@ module Make (Inputs : Inputs_intf) = struct
     ; transaction_pool_diffs:
         Transaction_pool_diff.t Envelope.Incoming.t Linear_pipe.Reader.t
     ; snark_pool_diffs:
-        Snark_pool_diff.t Envelope.Incoming.t Linear_pipe.Reader.t }
+        Snark_pool_diff.t Envelope.Incoming.t Linear_pipe.Reader.t
+    ; online_status: [`Offline | `Online] Broadcast_pipe.Reader.t }
   [@@deriving fields]
 
+  [%%if
+  consensus_mechanism = "proof_of_stake"]
+
+  [%%inject
+  "block_window_duration_ms", block_window_duration]
+
+  [%%inject
+  "delta", delta]
```

Comment:
you should use `Constants` instead of injecting here (I think you can avoid the %%if in that case too!)

Reply:
I can try to do that. I think including the `consensus` library into coda_networking will cause some cyclic dependencies. I may put the constants in another module if that's the case.